src/sts.rs

```diff
@@ -0,0 +1,278 @@
+//! The AWS STS API.
+//!
+//! If you're using the service, you're probably looking for [StsClient](struct.StsClient.html).
+#![allow(unused_imports)]
+
+include!(concat!(env!("OUT_DIR"), "/sts.rs"));
+
+pub use self::credential::{
+    StsSessionCredentialsProvider,
+    StsAssumeRoleSessionCredentialsProvider,
+    StsWebIdentityFederationSessionCredentialsProvider,
+    NewAwsCredsForStsCreds,
+    };
+
+mod credential {
+    use std::error::Error;
+    use std::fmt;
+
+    use hyper::Client;
+    use chrono::*;
+
+    use ::{AwsCredentials, ProfileProvider, CredentialsError,
+        ProvideAwsCredentials, DispatchSignedRequest};
+    use super::{AssumeRoleRequest, AssumeRoleResponse, AssumeRoleError,
+        AssumeRoleWithSAMLRequest, AssumeRoleWithSAMLResponse, AssumeRoleWithSAMLError,
+        AssumeRoleWithWebIdentityRequest, AssumeRoleWithWebIdentityResponse, AssumeRoleWithWebIdentityError,
+        DecodeAuthorizationMessageRequest, DecodeAuthorizationMessageResponse, DecodeAuthorizationMessageError,
+        GetCallerIdentityRequest, GetCallerIdentityResponse, GetCallerIdentityError,
+        GetFederationTokenRequest, GetFederationTokenResponse, GetFederationTokenError,
+        GetSessionTokenRequest, GetSessionTokenResponse, GetSessionTokenError,
+        StsClient};
+
+    pub const DEFAULT_DURATION_SECONDS: i32 = 3600;
+    pub const DEFAULT_ROLE_DURATION_SECONDS: i32 = 900;
+
+    pub trait NewAwsCredsForStsCreds {
```

Comment:
These are public constants in a private module. They are not re-exported.
The default durations are lifted from the java sdk, but also match the published defaults on the STS API itself.
Those defaults are documented in the methods on the generated StsClient.